### ISSUE TITLE
miou result can not achieve 74% 

### ISSUE THREAD

**Issue by tim120526** [speedinghzl/Pytorch-Deeplab#24]
HI! @speedinghzl , thanks for you code!
I used the VOC12_scenes_20000.pth model you provided to run the test code. I only change the path to load data, any other thing was changed in my operation. The Miou results  achieved  0.693386318233 in the single scale model , when i use the multi scale model you provided , the result is also 69%. Can you help me ? or give me some idea about it. 
Thank you very much! Good luck to u.

**Comment by speedinghzl** [comment]
What's the version of your Pytorch? I trained this model with pytorch==0.2, the reason may be the change of the upsampling operator.  But I have no time to figure it out. If you find a reason, please tell me. Thank you.

P.s. I refer you to use our new pytorch Implementations for DeeplabV3 and PSPNet. https://github.com/speedinghzl/pytorch-segmentation-toolbox

**Comment by tim120526** [comment]
My version is 0.4.1, I have not found the reason, I will tell you reason if i find it. Thanks a lot . I have another question to ask you,emm.... .I want to train my model with imagenet_pretrained model rather than coco_pretrained model in order to do experiment about my work, I found that if i load the resnet101_imagenet_pretrained_model pytorch provides, the loss in your code will be nan..., Do u have any idea about it ? Thanks a lot! @speedinghzl

**Comment by speedinghzl** [comment]
@tim20120526 Thanks. You could double check the parameters loading of imagenet-pretrain model, then reduce the learning rate until the loss decrease smoothly.

**Comment by tim120526** [comment]
Thanks a lot. @speedinghzl . I have found the reason that the different preprocess way between pytorch and caffe.

**Comment by ymli39** [comment]
Hi @tim20120526 , I used pytorch version of 0.4.0 and my IOU remains around 0.58 around 20000 iterations. I used batch size of 16 and max_iter of 100000 of i_iter, because if I set at 20000, the model will terminate around 666 b_iters. I was wondering what is your parameter for training and what is the issue that cause the lower performance?
Any help could be really appreciated...

**Comment by tim120526** [comment]
> Hi @tim20120526 , I used pytorch version of 0.4.0 and my IOU remains around 0.58 around 20000 iterations. I used batch size of 16 and max_iter of 100000 of i_iter, because if I set at 20000, the model will terminate around 666 b_iters. I was wondering what is your parameter for training and what is the issue that cause the lower performance?
> Any help could be really appreciated...

I implement it in pytorch 0.4.1 and python 3.6. And I only change the path to load data in the code.

**Comment by yzou2** [comment]
> Thanks a lot. @speedinghzl . I have found the reason that the different preprocess way between pytorch and caffe.

Hi @tim20120526, did you find the reason why there is a validation accuracy difference between your validation and the officially reported performance? Any help will be appreciated. I have the issue that the fine-tuned single-scale VOC12_scenes_20000.pth can only achieved mIoU of 71.1, rather than 74. I used pytorch version of 1.0.0 and only change the path to load data in the train code.
